### PR TITLE
Add d3

### PR DESCRIPTION
**Pull request by sannykr**

### CODE REVIEW

**Comment by dougalsutherland** [comment]
See discussion at https://github.com/conda-forge/staged-recipes/pull/4044#issuecomment-332019807.

**Comment by sannykr** [comment]
@conda-forge/staged-recipes : Our discussion about non python packages was resolved. Could you please review and merge. It fails only for python35 on window due to some system issues.

File: recipes/d3/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+{% set name = "d3" %}
+{% set version = "4.10.2" %}
+{% set sha256 = "eaa296738d4ec6bc49ebfe6635284972b38db5f146ba9f5778af45b6306044ae" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/d3/d3/archive/v4.10.2.tar.gz
```
Comment (dougalsutherland) [inline recipes/d3/meta.yaml]: Same as the others, template this.

File: recipes/d3/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  noarch: True
```
Comment (dougalsutherland) [inline recipes/d3/meta.yaml]: Again, `noarch: generic`.

File: recipes/d3/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  script: npm install -g .
+
+requirements:
+  build:
+    - nodejs
+  run:
+    - nodejs
+    
+test:
+  imports:
```
Comment (dougalsutherland) [inline recipes/d3/meta.yaml]: Again, add a meaningful test.

File: recipes/d3/meta.yaml [hunk 4]
```diff
@@ -22,7 +22,8 @@ requirements:
     - nodejs
```
Comment (isuruf) [inline recipes/d3/meta.yaml]: Is this needed?

**Comment by sannykr** [comment]
@conda-forge/staged-recipes : Could you please review and merge.

File: recipes/d3/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  script: npm install -g .
+
+requirements:
+  build:
+    - nodejs
+  run:
+    - nodejs
+test:
+  commands:
+    - npm list -g d3
```
Comment (isuruf) [inline recipes/d3/meta.yaml]: Can you have a look at https://github.com/conda-forge/staged-recipes/pull/4047 and do the same here?

**Comment by sannykr** [comment]
@conda-forge/staged-recipes : Please let me know if you have more suggestions.

**Comment by sannykr** [comment]
@conda-forge/staged-recipes : Could you please review and merge.

**Pull request merged**